src/components/mediaLibraryCreator/mediaLibraryCreator.js

```diff
@@ -169,24 +188,21 @@ define(['loading', 'dialogHelper', 'dom', 'jQuery', 'components/libraryoptionsed
     }
 
     function initLibraryOptions(dlg) {
-        libraryoptionseditor.embed(dlg.querySelector('.libraryOptions')).then(function () {
+        libraryoptionseditor.embed(dlg.querySelector('.libraryOptions')).then(() => {
             $('#selectCollectionType', dlg).trigger('change');
             onToggleAdvancedChange.call(dlg.querySelector('.chkAdvanced'));
         });
     }
 
-    function editor() {
-        this.show = function (options) {
-            return new Promise(function (resolve, reject) {
+export class editor {
+    constructor() {
+        this.show = options => {
```

Comment:
This can be moved into a property of the class, no need to define it in the constructor.